esp/services/ws_workunits/ws_workunitsHelpers.cpp

```diff
@@ -1975,7 +2038,32 @@ void WsWuInfo::getWorkunitThorLog(const char* fileName, MemoryBuffer& buf, const
     if (!rFile)
         throw MakeStringException(ECLWATCH_CANNOT_OPEN_FILE,"Cannot open file %s.",fileName);
 
-    readWorkunitLog(rFile, buf, outFile);
+    if (debugImportedStr.length()) //WU imported from a ZAP report
+        getWorkunitLogFromZAPFile(rFile, fileName, buf, outFile);
+    else
+        readWorkunitLog(rFile, buf, outFile);
+}
+
+void WsWuInfo::getWorkunitLogFromZAPFile(IFile* iFile, const char* fileName, MemoryBuffer& buf, const char* outFile)
+{
+    //This is a WU from ZAP report. The thor log only contains the log lines related to this WU.
+    //The whole log file should be sent back. 
+    if (!isEmptyString(outFile))
+    {
+        OwnedIFile oFile = createIFile(outFile);
+        if (!oFile)
+            throw MakeStringException(ECLWATCH_CANNOT_OPEN_FILE, "Cannot open %s.", outFile);
+
+        copyFile(oFile, iFile);
```

Comment:
Currently getWorkunitLogFromZAPFile is both writing to the file and copying to buffer.
There's a:
```
return;
```
missing here

Reply:
Will add back.